src/libasr/lsp.cpp

```diff
@@ -344,6 +344,14 @@ int get_errors(const std::string &infile, CompilerOptions &compiler_options)
     return 0;
 }
 
+bool is_id_chr(const char c)
+{
+    return (('a' <= c) && (c <= 'z'))
+        || (('A' <= c) && (c <= 'Z'))
+        || (('0' <= c) && (c <= '9'))
+        || (c == '_');
```

Comment:
There doesn't seem to be a single standard function to do that, so the above is probably good. The only other way I can think of is:
```suggestion
    return std::isalnum(c) || c == '_';
```
But it's still not a single function, so the original code seems more explicit / readable.